src/app/track/track.component.ts

```diff
@@ -1,346 +1,476 @@
+                zoom: 15
+            })
         });
-    this.layers.push(pointLayer);
-    const olMap = new Map({
-      layers: this.layers,
-      target: 'trackMapView',
-      view: new View({
-        maxZoom: 22,
-        center: points[0] || fromLonLat([9.1797, 48.7784]),
-        zoom: 15
-      })
-    });
-
-  }
-
-  exportGPX() {
-    this.isExporting = true;
-    let fileContents = '<?xml version="1.0"?>\n' +
-      '<gpx version="1.1" creator="openbikesensor.hlrs.de">\n';
-    fileContents += '<trk>\n';
-    fileContents += '<trkseg>\n';
-    let nameCounter = 1;
-    for (const p of this.trackData.points) {
-      fileContents += '<trkpt  lat="' + p.latitude + '" lon="' + p.longitude + '">\n' +
-        '<name>' + nameCounter + '</name>\n';
-      if (p.flag != null && p.flag > 0) {
-        fileContents += '<sym>Flag</sym>\n';
-      }
-      fileContents += '<cmt> d1="' + p.d1 + '" d2="' + p.d2 + '" </cmt>\n';
-      fileContents += '</trkpt>\n';
-      nameCounter++;
-    }
-    fileContents += '</trkseg>\n';
-    fileContents += '</trk>\n';
-    fileContents += '</gpx>';
-    const filename = 'track.gpx';
-    const filetype = 'text/plain';
-    const a = document.createElement('a');
-    const dataURI = 'data:' + filetype +
-      ';base64,' + btoa(fileContents);
-    a.href = dataURI;
-    a['download'] = filename;
-    const e = document.createEvent('MouseEvents');
-    // Use of deprecated function to satisfy TypeScript.
-    e.initMouseEvent('click', true, false,
-      document.defaultView ?? window, 0, 0, 0, 0, 0,
-      false, false, false, false, 0, null);
-    a.dispatchEvent(e);
-    a.remove();
-    this.isExporting = false;
-  }
-  exportCSV() {
-    this.isExporting = true;
-    let fileContents = 'Date;Time;Latitude;Longitude;Course;Speed;Right;Left;Confirmed;insidePrivacyArea\n';
-    for (const p of this.trackData.points) {
-      fileContents += [p.date, p.time, p.latitude, p.longitude, p.course, p.speed, p.d1, p.d2, p.flag, p.private].map(
-        (value) => {
-          if (value == null) {
-            return '';
-          } else if (typeof value === 'number') {
-            return String(value);
-          } else if (typeof value === 'boolean') {
-            return value ? '1' : '0';
-          } else if (typeof value === 'string' && /[^a-zA-Z0-9_.,: -]/.test(value)) {
-            // properly escape strings that contain quotes, semicolons, or
-            // anything else that is weird
-            return JSON.stringify(value);
-          } else {
-            // unproblematic string can be returned as-is
-            return value;
-          }
+
+        const layerSwitcher = new LayerSwitcher({
+            groupSelectStyle: 'children',
+            startActive: true,
+            activationMode: 'click',
+            reverse: false,
+
+
+        });
+        olMap.addControl(layerSwitcher);
+
+
+        function pointStyleFunction(feature, resolution) {
```

Comment:
Is there a reason why this is written as `function x`, and everything else als `const x = function`?

Reply:
Hmm not really. Looks like "new to TypeScript & Angular" :) 
What is your preferred way?

Reply:
`function x` is the more performant way, because then the function is only created once, even if multiple instances of the class are created.

Binding it to a `const` is only necessary if you want to pass functions as arguments into higher-order functions.